app/javascript/bundles/RestaurantsApp/components/AddReview.jsx

```diff
@@ -0,0 +1,114 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Formsy from 'formsy-react';
+import TextInput from '../../Forms/components/TextInput';
+import StarsRatingFilter from "../../StarsRating/components/StarsRatingFilter"
+
+
+export default class AddReviewForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isFormValid: false,
+            isFormSubmitted: false,
+            userRating: 0,
+            isAddingReview: false,
+        }
+        this.submit = this.submit.bind(this);
+        this.setValid = this.setValid.bind(this);
+        this.setInvalid = this.setInvalid.bind(this);
+
+    }
+
+    static propTypes = {
+        restaurant: PropTypes.object.isRequired,
+        onClose: PropTypes.func.isRequired,
+        onAdd: PropTypes.func.isRequired,
+    };
+
+    getInitialState() {
+        return {
+            isFormValid: false
+        }
+    }
```

Comment:
This method is probably redundant. You have something like this in the constructor.